lib/realtime/tenants/connect/backoff.ex

```diff
@@ -0,0 +1,45 @@
+defmodule Realtime.Tenants.Connect.Backoff do
```

Comment:
Backoff typically means a gradually increasing delay between retry attempts. Perhaps a more accurate name would be Throttler or RateLimiter.

For the config option, instead of `connect_backoff_limit`, it could be `connect_throttle_limit`, `connect_throttle_threshold` (it is a threshold on the average), or similar.